internal/service/handlers/join_program.go

```diff
@@ -0,0 +1,96 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/rarimo/geo-auth-svc/internal/data"
+	"github.com/rarimo/geo-auth-svc/internal/jwt"
+	"github.com/rarimo/geo-auth-svc/internal/service/requests"
+	"gitlab.com/distributed_lab/ape"
+	"gitlab.com/distributed_lab/ape/problems"
+)
+
+func JoinProgram(w http.ResponseWriter, r *http.Request) {
+	req, err := requests.NewJoinProgram(r)
+	if err != nil {
+		Log(r).WithError(err).Debug("Bad request")
+		ape.RenderErr(w, problems.BadRequest(err)...)
+		return
+	}
+
+	log := Log(r).WithFields(map[string]any{
+		"user.nullifier":    req.Data.ID,
+		"user.anonymous_id": req.Data.Attributes.AnonymousId,
+	})
+
+	nullifier := req.Data.ID
+	anonymousID := req.Data.Attributes.AnonymousId
+
+	claim := Claim(r)
+	if claim == nil || claim.Type != jwt.AccessTokenType || claim.Nullifier != nullifier {
```

Comment:
Use `pkg/auth.Authenticates`, add a custom `Grant` or modify existing if you wish

Reply:
pkg/auth.Authenticates is used for external validation, for this service there is an internal structure for claim. I suggest leaving it as it is